cmd/influxd/run.go

```diff
@@ -71,6 +72,16 @@ func execRun(args []string) {
 		}
 		log.Printf("data node #%d listening on %s", s.ID(), config.DataAddr())
 
+		// Spin up the collectd server
+		if config.Collectd.Enabled {
+			c := config.Collectd
+			cs := collectd.NewServer(s, c.TypesDB)
+			cs.Database = c.Database
+			err := collectd.ListenAndServe(cs, c.ConnectionString(config.BindAddress))
```

Comment:
I'm unsure that we can match the signature 100%.  ListenAndServer for TCP/UDP takes a single argument, just a address.  For us, we require a lot of server setup so I think we need to pass at least 2 arguments.

The reality is someone could just as easily do this:
```
err := collectd.ListenAndServe(&Server{}, addr)
```
and be the same as 
```
s := Server{}
s.ListenAndServer(addr)
```